src/simulator/player.py

```diff
@@ -196,14 +196,17 @@ def _perform_action(self, action, deck):
         :param deck: The deck object to pull cards from
         :return:
         """
-        if action == PlayerAction.STAY:
-            self._stay()
-        elif action == PlayerAction.HIT:
-            self._hit(deck)
-        elif action == PlayerAction.SPLIT:
-            self._split(deck)
-        elif action == PlayerAction.DOUBLE:
-            self._double(deck)
+        if self.curr_hand_ndx < len(self.hands):
+            if action == PlayerAction.STAY:
+                self._stay()
+            elif action == PlayerAction.HIT:
+                self._hit(deck)
+            elif action == PlayerAction.SPLIT:
+                self._split(deck)
+            elif action == PlayerAction.DOUBLE:
+                self._double(deck)
+        else:
+            raise BlackjackException("You can't perform another since you have a blackjack")
```

Comment:
I'd be down to make this an `elif self.hands[self.curr_hand_ndx - 1].blackjack:`, then throw a `ValueError' for `else`, just so that we're not including any potential bugs where the player's `curr_hand_ndx` skips over something as blackjack.